tools/codegen/api/cpp.py

```diff
@@ -191,9 +195,89 @@ def argument(a: Union[Argument, TensorOptionsArguments, ThisArgument]) -> CppArg
     else:
         assert_never(a)
 
-def group_arguments(
-    func: FunctionSchema, *, method: bool = False
-) -> Sequence[Union[Argument, TensorOptionsArguments, ThisArgument]]:
+@dataclass(frozen=True)
+class CppSignature:
+    arguments: Sequence[Union[Argument, TensorOptionsArguments, ThisArgument]]
+
+    def cpp_arguments(self) -> Sequence[CppArgument]:
+        return list(map(argument, self.arguments))
+
+    # Return arguments as a comma separated list, i.e. like they would be in a C++
+    # function signature. Include default values for arguments.
+    def cpp_arguments_str(self) -> str:
+        return ', '.join(map(str, self.cpp_arguments()))
+
+    # Return a copy of the CppSignature with the `this` argument removed
+    def exclude_this(self) -> 'CppSignature':
+        return CppSignature(
+            arguments=[a for a in self.arguments if not isinstance(a, ThisArgument)]
+        )
+
+    # Return a copy of the CppSignature with all default values removed from the signature
+    def no_defaults(self) -> 'CppSignature':
```

Comment:
Reasoning similar to `exclude_this` applies here

Reply:
I made it into a parameter